From efba627ae57e3b328967fe231002e9904942fe8d Mon Sep 17 00:00:00 2001
From: Terry Zhou <bjzhou@marvell.com>
Date: Tue, 1 Mar 2016 09:23:38 +0800
Subject: [PATCH 0139/1345] dts: pinctrl: a3700: enable sdhc related pins in
 the dts nodes

commit  d3efbb8812a6bd45ca47889c6298e33bc38763d2 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

set sdhci0 to mmc mode
set sdhci1 to sdio mode

Change-Id: Ief71d0e5451e8455e6281528ef9a3484e9a4c7dd
Signed-off-by: Terry Zhou <bjzhou@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27952
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-db.dts |    3 +++
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi   |   10 ++++++++++
 2 files changed, 13 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index 3566b65..f1c688d 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -162,6 +162,9 @@
 	mmc-hs400-1_8v;
 	marvell,pad-type = "fixed-1-8v";
 	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&mmc_pins>;
 };
 
 &sdhci1 {
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index fd7c8f2..60823ec 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -273,6 +273,11 @@
 					marvell,pins = "mpp15", "mpp18";
 					marvell,function = "spi";
 				};
+
+				mmc_pins: mmc-pins {
+					marvell,pins = "mpp2";
+					marvell,function = "mmc";
+				};
 			};
 
 			sgateclk: south-bridge-gate-clk@18014 {
@@ -284,6 +289,11 @@
 			spinctrl: south-bridge-pinctrl@18830 {
 				compatible = "marvell,armada-3700-sb-pinctrl";
 				reg = <0x18830 0x4>;
+
+				sdio_pins: sdio-pins {
+					marvell,pins = "mpp2";
+					marvell,function = "sdio";
+				};
 			};
 
 			usb3: usb@58000 {
-- 
1.7.9.5

